src/chunkserver/trash.cpp

```diff
@@ -121,13 +121,15 @@ int Trash::RecycleCopySet(const std::string &dirPath) {
                      << trashPath_;
         return -1;
     }
-
-    if (0 != localFileSystem_->Rename(dirPath, dst)) {
-        LOG(ERROR) << "rename " << dirPath << " to " << dst << " error";
-        return -1;
+    {
+        LockGuard lg(mtx_);
+        if (0 != localFileSystem_->Rename(dirPath, dst)) {
+            LOG(ERROR) << "rename " << dirPath << " to " << dst << " error";
+            return -1;
+        }
+        uint32_t chunkNum = CountChunkNumInCopyset(dst);
```

Comment:
CountChunkNumInCopyset does not contain wal file?

Reply:
This issue is left to fix by future PR

Reply:
There is `expiredAfterSec_` seconds interval between `RecycleCopySet()` and `RecycleChunksAndWALInDir()`，so what causes concurrency issues?. See [NeedDelete](https://github.com/opencurve/curve/blob/ad79946fa0b5463278155774ae16282526885a01/src/chunkserver/trash.cpp#L198)